ddl/ddl_api.go

```diff
@@ -6400,6 +6455,129 @@ func (d *ddl) AlterTableAttributes(ctx sessionctx.Context, ident ast.Ident, spec
 
+func (d *ddl) AlterTableStatsOptions(ctx sessionctx.Context, ident ast.Ident, spec *ast.StatsOptionsSpec) error {
+	schema, tb, err := d.getSchemaAndTableByIdent(ctx, ident)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	meta := tb.Meta()
+	statsOptions := model.NewStatsOptions()
+	if !spec.Default {
+		if spec.StatsOptions == "" && meta.StatsOptions != nil {
+			statsOptions = meta.StatsOptions
+		} else if spec.StatsOptions != "" {
+			attrBytes := []byte(spec.StatsOptions)
+			var attrs map[string]string
+			err = json.Unmarshal(attrBytes, &attrs)
+			if err != nil {
+				return errors.Wrapf(err, "Stats options should be in json format")
+			}
+			hasBuckets, hasTopn, hasSampleRate, hasColChoice, hasColList := false, false, false, false, false
+			for key, val := range attrs {
+				switch strings.ToUpper(key) {
+				case "BUCKETS":
+					hasBuckets = true
+					if strings.ToLower(val) == "default" {
+						statsOptions.Buckets = uint64(0)
+					} else {
+						buckets, err0 := strconv.ParseUint(val, 0, 64)
+						if err0 != nil {
+							return fmt.Errorf("%w: %s", errors.New("BUCKETS should be a unit"), val)
+						}
+						statsOptions.Buckets = buckets
+					}
+				case "TOPN":
+					hasTopn = true
+					if strings.ToLower(val) == "default" {
+						statsOptions.TopN = uint64(0)
+					} else {
+						topn, err0 := strconv.ParseUint(val, 0, 64)
+						if err0 != nil {
+							return fmt.Errorf("%w: %s", errors.New("TOPN should be a unit"), val)
+						}
+						statsOptions.TopN = topn
+					}
+				case "SAMPLE_RATE":
+					hasSampleRate = true
+					if strings.ToLower(val) == "default" {
+						statsOptions.SampleRate = 0.0
+					} else {
+						sampleRate, err0 := strconv.ParseFloat(val, 64)
+						if err0 != nil {
+							return fmt.Errorf("%w: %s", errors.New("SAMPLE_RATE should be a float"), val)
+						}
+						statsOptions.SampleRate = sampleRate
+					}
+				case "COL_CHOICE":
+					hasColChoice = true
+					switch strings.ToLower(val) {
+					case "all":
+						statsOptions.ColumnChoice = model.AllColumns
+					case "list":
+						statsOptions.ColumnChoice = model.ColumnList
+					case "predicate":
+						statsOptions.ColumnChoice = model.PredicateColumns
+					case "default":
+						statsOptions.ColumnChoice = model.AllColumns
+					default:
+						return fmt.Errorf("%w: %s", errors.New("COL_CHOICE should be one of all|list|predicate"), val)
+					}
+				case "COL_LIST":
+					hasColList = true
+					if strings.ToLower(val) == "default" {
+						statsOptions.ColumnList = []model.CIStr{}
+					} else {
+						cols := strings.Split(val, ",")
+						colList := make([]model.CIStr, 0, len(cols))
+						for _, colStr := range cols {
+							colName := strings.TrimSpace(colStr)
+							colList = append(colList, model.NewCIStr(colName))
+						}
+						statsOptions.ColumnList = colList
+					}
+				default:
+					return errors.Trace(errors.New("unknown table stats option"))
+				}
+			}
+			// merge old meta if not set at this time
```

Comment:
We'd better make it a complete sentence.
```suggestion
			// Merge old meta if not set at this time.
```